opendrift/models/basemodel/__init__.py

```diff
@@ -2225,14 +2225,18 @@ def run(self,
         # Remove any elements scheduled for deactivation during last step
         self.remove_deactivated_elements()
 
-        if export_buffer_length is None:
-            pass  # TODO - do this for self.result
-        else:  # If output has been flushed to file during run, we
-            # need to reimport from file to get all data in memory
-            del self.environment
-            if hasattr(self, 'environment_profiles'):
-                del self.environment_profiles
-            self.result = xr.open_dataset(outfile)
+        if outfile is not None:
+            ext = os.path.splitext(outfile)[1].lower()
+            if ext in [".nc", ".nc4", ".netcdf"]:
+                # NetCDF: open with xarray as before
+                self.result = xr.open_dataset(outfile)
+            elif ext == ".parquet":
+                # Parquet: let the parquet exporter set self.result in its close()
+                # (we do nothing here)
+                pass
+            else:
+                # Fallback: try normal xarray open
+                self.result = xr.open_dataset(outfile)
```

Comment:
The lines 2230-2239 also seem a bit overly complicated. Could this not be simply:
```
if outfile is not None:
    ext = os.path.splitext(outfile)[1].lower()
    if ext == ".parquet":
         pass  # let the parquet exporter set self.result in its close()
    else:
         # reimport from file to get the complete simulation (lazily) in memory
         self.result = xr.open_dataset(outfile)  
```

Reply:
Also I am note completely sure about replacing `export_buffer_length is None` with `outfile is None`,
as one could have an outfile but still set `export_buffer_length` to `None` to get just a single write at the end.
In this case it would be unnecessary to reimport from that file, as the complete simulation is already in memory.
On the other hand, it might perhaps be ok for consistency.